src/QMCHamiltonians/QMCHamiltonian.cpp

```diff
@@ -1121,4 +1121,182 @@ RefVectorWithLeader<OperatorBase> QMCHamiltonian::extract_HC_list(const RefVecto
   return HC_list;
 }
 
+QMCHamiltonian::FullPrecRealType QMCHamiltonian::evaluateIonDerivsDeterministicFast(ParticleSet& P,
+                                                                                    ParticleSet& ions,
+                                                                                    TrialWaveFunction& psi_in,
+                                                                                    ParticleSet::ParticlePos& dEdR,
+                                                                                    ParticleSet::ParticlePos& wf_grad)
+{
+  ScopedTimer evaluatederivtimer(*timer_manager.createTimer("FastDeriv::evaluateIonDerivsFast"));
+  if (!psi_wrapper_.isInitialized())
+  {
+    psi_in.initializeTWFFastDerivWrapper(P, psi_wrapper_);
+    psi_wrapper_.finalizeConstruction();
+  }
+  P.update();
+  //resize everything;
+  int ngroups = psi_wrapper_.numGroups();
```

Comment:
`const int` to prevent unintended changes.

Reply:
Done.